pkg/models/sigv4.go

```diff
@@ -0,0 +1,78 @@
+package models
+
+import (
+	"errors"
+	"net/http"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws/credentials"
+	"github.com/aws/aws-sdk-go/aws/credentials/stscreds"
+	"github.com/aws/aws-sdk-go/aws/session"
+	v4 "github.com/aws/aws-sdk-go/aws/signer/v4"
+)
+
+type AuthType string
+
+const (
+	Keys        AuthType = "keys"
+	Credentials AuthType = "credentials"
+	ARN         AuthType = "arn"
+)
+
+type Sigv4Middleware struct {
```

Comment:
Isn't it better to name it "SigV4Middleware" to be consistent with how SigV4 is usually written?
```suggestion
type SigV4Middleware struct {
```